### ISSUE TITLE
User info is not being correctly shown in Jira when merging - is this possible?

### ISSUE THREAD

**Issue by leighgallagher** [atlassian/github-for-jira#1816]
I am looking to see if there is a way to show the user who merged into the Jira ticket. As you can see in the image below, the Author is shown as 'User' - is there a way around this?

<img width="1429" alt="image-20230102-090042" src="https://user-images.githubusercontent.com/100769506/210237090-7433732a-415f-48bc-b0e0-f579d31df492.png">

**Comment by krazziekay** [comment]
Hi @leighgallagher , can you please check if this user's GitHub account is the same as his/her Jira's account? 
Due to a recent internal change, Jira now shows the user information only if the user's GitHub email is the same as his/her Atlassian email. This has been well explained in the [FAQ](https://github.com/atlassian/github-for-jira/blob/main/docs/FAQs.md#q-why-cant-i-see-the-author-reviewer-names-andor-icons-in-my-jira) section.

**Comment by roufi3** [comment]
Hey folks
What if the email address matches and it's still not shown anymore?

**Comment by birdsarah** [comment]
As best as I can tell the only matches that are currently showing up are when a user's github handle / username matches the email username e.g. my github handle is `birdsarah` my jira email address is *not* `birdsarah@mozilla.com`. 

My jira email address (authentication is by SSO) is in my github profile, though, and is currently set to my primary email address. 

I have tried various combinations on Github of making it my default address, making it public, and putting it in a github commit message. None of these seem to work. 

Can the [FAQ](https://github.com/atlassian/github-for-jira/blob/main/docs/FAQs.md#q-why-cant-i-see-the-author-reviewer-names-andor-icons-in-my-jira) section you linked add more specificity than "the email address associated with your GitHub account should match the address in your Jira account"

The github settings where I see options are
* https://github.com/settings/profile (set a public email address)
* https://github.com/settings/emails (add multiple emails, set a primary email address, set other options for email)

**Comment by RichardDavies** [comment]
I agree with @birdsarah 	that this new seemingly arbitrary requirement is very confusing and frustrating to a user. My Github account has two email addresses associated with it at https://github.com/settings/emails, and one of them is the exact same email address as my Atlasian email. However, regardless of whether or not that email is set as my "primary" Github email, I no longer see the Github user info in Jira.

I can see other commit and pull request details in Jira so the accounts are obviously connected. Showing the user information should "just work" as it previously did as long as the accounts are properly connected. The new matching email address requirement is too brittle and arbitrary and not working correctly.

**Comment by birdsarah** [comment]
I dug into this a bit more. To get my avatar and name to correctly show in Jira, I need to make my Jira email, my *public* email address on github. This does not backfill past information but new PRs that are opened will now correctly have my avatar.

The octokit library that github-for-jira uses, provides https://octokit.github.io/rest.js/v19#users-list-emails-for-authenticated-user which could be used for non-public email address matching if github-for-jira has the `user:email` scope.

**Comment by tjlytle** [comment]
We'd also find it useful if this worked for _every_ email and not just the public email.

**Comment by rachellerathbone** [comment]
Hey all. Unfortunately, this regression is the result of a upstream team changing the way they store data. A team mate of mine is working on a doc to outline what can be done to remediate this via your GH settings, until we find a longer-term strategy.

I'll send the link through once it's ready.

**Comment by rachellerathbone** [comment]
Our SUPPORT docs have been updated with more info and a temporary workaround https://github.com/atlassian/github-for-jira/blob/main/SUPPORT.md#unmapped-users Unfortunately, this is not the most ideal solution but it is the best we can do at this point in time. We're trying to coordinate with the team that made the changes to see if there is a way to return this functionality for all users.

**Comment by JoppeDC** [comment]
This is a pretty annoying issue, as not everyone can just set their email address to be their public Github address. Also, the integration is using some old profile picture (~10+ years old) from who knows where. Its not picking up the GIthub profile picture at all.
In another issue i read that this uses the Gravatar profile picture, but it appears that i do not have a gravatar account so thats also not the case.

Pretty bad issue, business whise.

**Comment by augustnmonteiro** [comment]
Is there any update on this? I'm facing the same issue.

It is pretty anoying and we can't just ask all the devs to change their github public email.

what if there was a field in the users profile for github username and you guys could use that as a match?

**Comment by deemsk** [comment]
This issue also affects my team badly. It's been almost a years since this issue was opened. Are there any news? How much longer the fix will take?

**Comment by stguitar** [comment]
@rachellerathbone yea, still an issue, and the shared workaround is not really even close to plausible.  

@birdsarah I was intrigued by your suggestion of using the API in someway.  Can you expand on that a bit?

**Comment by watk** [comment]
Any updates on this?

**Comment by Radek-Pysny** [comment]
At least this issue is not closed yet, but there could be done some progress @rachellerathbone. It would be best if this would be enabled also retrospectively, because it is very annoying. Could someone discuss this issue with upstream team, so it does not stuck forever...

**Comment by ZBishopM** [comment]
Good afternoon, I'm having this issue, tried to match email from github and jira's, but not solving. Do we have an update? :(

**Comment by Radek-Pysny** [comment]
@ZBishopM There are two points one might make an easy mistake. Namely, by thinking that it does any change retrospectively (no it changes only future code reviews, approvals, etc.) and by not having that email published (hidden email cannot be accessed via API).

**Comment by MiDoCisco** [comment]
@ZBishopM what I found confusing is that there are 2 places to indicate e-mail "private" - 

1. In the https://github.com/settings/emails - the "Keep my email addresses private"
2. In the profile https://github.com/YOUR_ID - you need to click "Edit Profile" and set the e-mail to non-empty value. This is what JIRA will be able to see. 

I believe that both places need to be set to public for the integration to match the users correctly in all cases.

**Comment by keir-nellyer** [comment]
I use the same GitHub account for personal & work, and so this issue is a bit of a pain for me. It means I have to set my work email to be displayed on my profile when I'd rather display my personal email address, or no email address at all.

**Comment by gavinclarkeuk** [comment]
This issue is a massive pain for us. I just wanted to highlight that it should be possible to retrieve a GH users organisational email information through the API if you have SAML/SSO set up. It is a slightly different API and I believe there is a different if you are using orgs or enterprise. Using these APIs this would avoid the nasty workaround of having make your email public.

Even without that it change, why not have JIRA show the github user id & avatar for cases when a match can't be found to an atlassian user. That would be a vast improvement for most people

**Comment by ryanbeymer-qgiv** [comment]
Is there a delay when adjusting these settings? 

I've been experimenting with a co worker and they have their email matching but it won't show up on new any new PR's.
